lib/compiler.js

```diff
@@ -670,51 +699,52 @@ function fixFileExtension(filepath, files, onlyScriptsExportingFunction) {
 
       if (!(otherFileExtension in getExcludedExtensions()) &&
         path.basename(f, otherFileExtension) == basename) {
-        if (!onlyScriptsExportingFunction)
-          results.push(otherFile);
-        else if (onlyScriptsExportingFunction &&
-          (typeof require.extensions[otherFileExtension]) === 'function') {
-          results.push(otherFile);
-        }
+        // if (!onlyScriptsExportingFunction)
+        results.push(otherFile);
+        //else if (onlyScriptsExportingFunction &&
+          //(typeof scriptExtensions[otherFileExtension]) === 'function') {
+          //results.push(otherFile);
+        //}
       }
     }
   });
   return (results.length > 0 ? results[0] : undefined);
 }
 
-function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
+function resolveAppScriptPath(rootDir, relativePath, resolveOptions,
```

Comment:
Please keep `resolveOptions` as the last function arg. I would consider converting `scriptExtensions` into a property in `resolveOptions`, to keep the number of function arguments reasonably short.